packages/datadog-plugin-ws/test/index.spec.js

```diff
@@ -0,0 +1,331 @@
+            expect(traces[0][0]).to.have.property('type', 'websocket')
+          })
+        })
+
+        it('should trace received messages when traceWebsocketMessagesEnabled is set to true', () => {
+          messageReceived = false
+          wsServer.on('connection', (ws) => {
+            ws.send('test message')
+          })
+          wsServer.on('message', (data) => {
+            expect(data.toString()).to.equal('test message')
+            expect(messageReceived).prototype.equal(true)
+          })
+
+          client.on('message', (data) => {
+            client.send(data)
+            expect(data.toString()).to.equal('test message')
+            messageReceived = true
+          })
+
+          return agent.assertSomeTraces(traces => {
+            expect(traces[0][0]).to.have.property('service', 'custom-ws-service')
+            expect(traces[0][0]).to.have.property('name', 'websocket.send')
+            expect(traces[0][0]).to.have.property('type', 'websocket')
+          })
+        })
+
+        it('should trace send messages when messages are not received', () => {
+          messageReceived = false
+          wsServer.on('connection', (ws) => {
+            ws.send('test message')
+          })
+          client.on('message', (data) => {
+            client.send(data)
+            expect(data.toString()).to.equal('test message')
+            messageReceived = true
+          })
+
+          return agent.assertSomeTraces(traces => {
+            expect(traces[0][0]).to.have.property('service', 'custom-ws-service')
+            expect(traces[0][0]).to.have.property('name', 'websocket.send')
+            expect(traces[0][0]).to.have.property('type', 'websocket')
+          })
+        })
+      })
+      describe('with WebSocket Messages Disabled', () => {
+        beforeEach(async () => {
+          await agent.load(['ws'], [{
+            service: 'custom-ws-service',
+            traceWebsocketMessagesEnabled: true
+          }])
+          WebSocket = require(`../../../versions/ws@${version}`).get()
+
+          wsServer = new WebSocket.Server({ port: clientPort })
+
+          client = new WebSocket(`ws://localhost:${clientPort}`)
+        })
+
+        afterEach(async () => {
+          clientPort++
+          agent.close({ ritmReset: false, wipe: true })
+        })
+
+        it('should not produce message spans when traceWebsocketMessagesEnabled is not set to true', () => {
+          wsServer.on('connection', (ws) => {
+            ws.send('test message')
+          })
+          messageReceived = false
+
+          client.on('message', (data) => {
+            expect(data.toString()).to.equal('test message')
+            messageReceived = true
+          })
+
+          return agent.assertSomeTraces(traces => {
+            expect(traces[0][0]).to.have.property('service', 'custom-ws-service')
+            expect(traces[0][0]).to.have.property('name', 'websocket.request')
```

Comment:
I believe this should be `websocket.send` or `websocket.receive`.

Reply:
in this case we are not producing message spans so it will be `web.request`